Assets/Scripts/Framework/Gameplay/Player/Interactor.cs

```diff
@@ -0,0 +1,113 @@
+using FishNet.Object.Synchronizing;
+using System.Collections.Generic;
+using System.Collections;
+using FishNet.Object;
+using UnityEngine;
+
+using DerailedDeliveries.Framework.Gameplay.Interactions;
+using DerailedDeliveries.Framework.InputParser;
+
+namespace DerailedDeliveries.Framework.Gameplay.Player
+{
+    /// <summary>
+    /// A class that is responsible for handling with in range Interactables for the player.
+    /// </summary>
+    [RequireComponent(typeof(CapsuleCollider))]
+    public class Interactor : NetworkBehaviour
+    {
+        [SerializeField]
+        private List<Interactable> _interactables;
+
+        [SerializeField]
+        private Interactable _interactingTarget;
+
+        [SerializeField]
+        private Transform _grabbingAnchor;
+
+        [SerializeField]
+        private float _cooldown = .2f;
+
+        private PlayerInputParser _inputParser;
+
+        [SyncVar(Channel = FishNet.Transporting.Channel.Reliable)]
+        private bool _isInteracting;
+
+        private bool _isOnCooldown;
+
+        private void Awake() => _inputParser = gameObject.GetComponent<PlayerInputParser>();
+
+        private void OnEnable() => _inputParser.OnInteract += UseInteractable;
+
+        private void OnDisable() => _inputParser.OnInteract -= UseInteractable;
+
+        private void OnTriggerEnter(Collider collider)
+        {
+            GameObject target = collider.gameObject;
+
+            if (target.TryGetComponent<Interactable>(out Interactable interactable))
```

Comment:
```suggestion
            if (target.TryGetComponent(out Interactable interactable))
```
This is redundant, type can already be retrieved by out keyword